Frameworks/Foundation/NSThread.mm

```diff
@@ -356,8 +356,15 @@ + (NSThread*)currentThread {
     NSThread* currentThread = [[self class] _threadObjectFromCurrentThread];
 
     if (currentThread == nil) {
-        currentThread = [NSThread new];
+        // Underlying thread created outside the NSThread APIs
+
+        // So create a new instance
+        currentThread = [[NSThread alloc] init];
         [currentThread _associateWithCurrentThread];
```

Comment:
Actually, it seems like this could be wrapped up into the current thread machinery. At file scope, a `thread_local StrongId<NSThread>`, which `_associateWith...` and `_threadObjectFrom...` could refer to. That way, its lifetime would be managed by the same thing that keeps track of its current-threadiness.